### ISSUE TITLE
Requested node configuration is not available

### ISSUE THREAD

**Issue by karlvirgil** [aws/aws-parallelcluster#282]
I am trying to submit a new job to the slurm scheduler. There are seven idle nodes (from jobs that just finished up).  I am requesting 10 nodes for the new job.  I get this error message:

sbatch: error: Batch job submission failed: Requested node configuration is not available

If I wait until all 7 idle nodes are removed from the scheduler and then submit the 10 node job, the scheduler will accept it and then slowly start launching the nodes to meet the request.

Here's my sbatch script:
#!/bin/bash
#SBATCH --job-name=ebyycncre
#SBATCH --partition=compute
#SBATCH --account=centos
#SBATCH --threads-per-core=1
#SBATCH --sockets-per-node=2
#SBATCH --cores-per-socket=18
#SBATCH --nodes=10
#SBATCH --ntasks=360
#SBATCH --ntasks-per-node=36
#SBATCH --mail-type=ALL
#SBATCH --mail-user=karl.meredith@fmglobal.com
#SBATCH --output=./s%j
#SBATCH --no-requeue

mpirun -np 360 fireFoam -parallel

And the output from sinfo:

PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
compute*     up   infinite      7   idle ip-10-15-20-[11,34,104,154,195,214-215]

**Comment by bwbarrett** [comment]
Unfortunately, we've never found a good solution to this problem.  Slurm has a mental model of a "cluster" that doesn't really fit with what we're trying to do with CfnCluster auto-scaling.  It assumes that nodes don't dynamically come and go (they may go "up" or "down", but there's a definite list of nodes that "belong" to the cluster).  We could potentially auto-add the entire subnet, but that would have other problems (like I'm not entirely sure the system would be well behaved if there were two clusters in the same subnet) and SLURM could be a bit more dynamic, but neither are likely to help you today.

Slurm will always be a little brittle with today's setup of CfnCluster.  It really only makes sense if you either run jobs serially (in which case our "phantom" queue will cause SLURM to accept the job).  If you want to scale down lower than the max job size you expect to have, then SLURM probably isn't the best batch scheduler to use with CfnCluster.  SGE is a bit more dynamic, probably due to its more GRID-friendly history.

We should document some of these tradeoffs better in our CfnCluster documentation.  I'm going to leave this issue open to track that effort.

**Comment by karlvirgil** [comment]
Okay.  Thanks for the explanation.  It'd be really nice to have better support for slurm, as it has a lot of really nice features over SGE.  

In the meantime, is it possible for me to switch my existing cfncluster instance from a slurm-based scheduler to an SGE-based scheduler without needing to create a new instance?  Is it as simple as changing 
```
scheduler = slurm
```
to 
```
scheduler = sge
```
in the .cfncluster/config file and then run
```
cfncluster update [instance]
```
?

**Comment by karlvirgil** [comment]
Sorry, but does anyone have any comments on the question I asked in my previous comment: "...is it possible for me to switch my existing cfncluster instance from a slurm-based scheduler to an SGE-based scheduler without needing to create a new instance?"

**Comment by bwbarrett** [comment]
`cfncluster update <cluster>` won't switch the scheduler.  Due to the way we use CloudFormation, CloudFormation doesn't consider the cluster type to be a configuration change.  Even if it did, the update wouldn't do what you want; for that change to have impact, it would have to terminate the master node and start a new one.  So your best bet is to terminate the existing cluster and create a new one.

**Comment by demartinofra** [comment]
This has been fixed since ParallelCluster 2.3.1.
